src/eth/evm/evm.rs

```diff
@@ -124,16 +124,17 @@ impl EvmInput {
     }
 
     /// Creates a transaction that was executed in an external blockchain and imported to Stratus.
+    ///
+    /// Successful external transactions executes with max gas and zero gas price to ensure we will have the same execution result.
     pub fn from_external_transaction(block: &ExternalBlock, tx: ExternalTransaction, receipt: &ExternalReceipt) -> Self {
-        let gas_limit = if_else!(receipt.is_success(), Gas::MAX, tx.0.gas.into());
         Self {
             from: tx.0.from.into(),
             to: tx.0.to.map_into(),
             value: tx.0.value.into(),
             data: tx.0.input.into(),
             nonce: Some(tx.0.nonce.into()),
-            gas_limit,
-            gas_price: tx.0.gas_price.expect("gas_price must be set for ExternalTransaction").into(), // XXX: how to handle transactions without gas price?
+            gas_limit: if_else!(receipt.is_success(), Gas::MAX, tx.0.gas.into()),
+            gas_price: if_else!(receipt.is_success(), Wei::ZERO, tx.0.gas_price.map_into().unwrap_or(Wei::ZERO)),
```

Comment:
I really wish those kind of macro did not exist, it's like bringing features to the language, and can lead to feature creep.

also you can always do something like this:
```
let a = if x > 5 { 10 } else { 7 };
```